metadata/metadata_test.go

```diff
@@ -205,7 +207,8 @@ func (s) TestFromIncomingContext(t *testing.T) {
 	)
 	// Verify that we lowercase if callers directly modify md
 	md["X-INCORRECT-UPPERCASE"] = []string{"foo"}
-	ctx := NewIncomingContext(context.Background(), md)
+	ctx := NewIncomingContext(ctxWithTimeout, md)
+	defer cancel()
```

Comment:
Ha, you need to call `cancel()` somewhere... but here you'll end up cancelling multiple times, which is a bit confusing, imo. I think it's better to move the context (which you can just call `ctx` and reassign when you pass it to NewIncomingContext) in each function. Sorry for the confusion on this.

Reply:
Sure, Moved context creation back to functions. 
Also renamed all ctxWithTimeout back to ctx.